samples/drivers/rtc/src/main.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024, Muhammad Waleed Badar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+
+const struct device *const rtc = DEVICE_DT_GET(DT_ALIAS(rtc));
+
+static int32_t set_date_time(const struct device *rtc)
+{
+	int32_t ret = 0;
+	struct rtc_time tm = {.tm_year = 2024 - 1900,
+			      .tm_mon = 7 - 1,
+			      .tm_mday = 28,
+			      .tm_hour = 3,
+			      .tm_min = 51,
+			      .tm_sec = 0};
+
+	ret = rtc_set_time(rtc, &tm);
+	if (ret < 0) {
+		printk("Cannot write date time: %d\n", ret);
+		return ret;
+	}
+	return ret;
+}
+
+static int32_t get_date_time(const struct device *rtc)
+{
+	int32_t ret = 0;
+	struct rtc_time tm;
+
+	ret = rtc_get_time(rtc, &tm);
+	if (ret < 0) {
+		printk("Cannot read date time: %d\n", ret);
+		return ret;
+	}
+
+	printk("Current RTC time: %04d-%02d-%02d %02d:%02d:%02d\n", tm.tm_year + 1900,
+	       tm.tm_mon + 1, tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec);
+
+	return ret;
+}
+
+int main(void)
+{
+	int32_t ret = 0;
+
+	/* Check if the RTC is available */
+	if (rtc == NULL) {
+		printk("Error: no device found\n");
+		return 0;
+	}
+
+	/* Check if the RTC is ready */
+	if (!device_is_ready(rtc)) {
+		printk("Device is not ready\n");
+		return 0;
+	}
+
+	/* Continuously read the current date and time from the RTC */
+	while (1) {
+		ret = get_date_time(rtc);
+		if (ret < 0) {
```

Comment:
Specific return value is not used, so this can be simplified to:
```
if (get_date_time(rtc)) {
        printk("Failed to read from rtc\n");
        return 0;
}
```
and even:
```
while (get_date_time(rtc) == 0) {
        k_sleep(K_MSEC(1000));
};
```

Reply:
I have made changes as you requested.